RxCocoa/Common/DelegateProxyFactory.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  DelegateProxyFactory.swift
+//  RxCocoa
+//
+//  Created by tarunon on 2017/06/17.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+#if !os(Linux)
+    
+#if !RX_NO_MODULE
+    import RxSwift
+#endif
+
+/**
+Define `DelegateProxy.init` for a specific object type.
+For example, in RxScrollViewDelegateProxy
+
+
+    class RxScrollViewDelegateProxy: DelegateProxy {
+        static var delegateProxyFactory = DelegateProxyFactory { (parentObject: UIScrollView) in
+            RxScrollViewDelegateProxy(parentObject: parentObject)
+        }
+    ...
+
+
+If need to extend them, chain `extended` after DelegateProxyFactory.init
+
+    class RxScrollViewDelegateProxy: DelegateProxy {
+        static var delegateProxyFactory = DelegateProxyFactory { (parentObject: UIScrollView) in
+                RxScrollViewDelegateProxy(parentObject: parentObject)
+            }
+            .extended { (parentObject: UITableView) in
+                RxTableViewDelegateProxy(parentObject: parentObject)
+            }
+    ...
+ 
+ 
+ */
+public class DelegateProxyFactory {
+    var factories: [ObjectIdentifier: ((AnyObject) -> AnyObject)]
```

Comment:
We use `_` conventions for private. This could be `private` as far as I can tell.